File: level2/타겟넘버.cpp
#include <string>
#include <vector>

using namespace std;

void getTarget(vector<int> numbers, int target, int s, int num, int &answer){
    if(s == numbers.size()){
        if(num == target){
            answer++;
        }
        return;
    }
    getTarget(numbers,target,s+1,num+numbers[s],answer);
    getTarget(numbers,target,s+1,num-numbers[s],answer);
}

int solution(vector<int> numbers, int target) {
    int answer = 0;
    getTarget(numbers,target,0,0,answer);
    return answer;
}

File: level2/타겟넘버.py
def solution(numbers, target):
    answer = 0
    
    stack = []
    stack.append([0,numbers[0]])
    stack.append([0,-1*numbers[0]])
    
    while stack:
        node = stack.pop(len(stack)-1)
        
        if node[0] == len(numbers)-1:
            if node[1] == target:
                answer += 1
            continue
            
        stack.append([node[0]+1,node[1]+numbers[node[0]+1]])
        stack.append([node[0]+1,node[1]-numbers[node[0]+1]])
    
    return answer
